Add Installation Guide  for PGMQ (#323)
tembo-io/pgmq: CONTRIBUTING.md

docker run -d --name postgres -e POSTGRES_PASSWORD=postgres -p 5432:5432 quay.io/tembo/pgmq-pg:latest
```

## PGXN Installation

PGXN is a centralized network to make it easier to install and manage PostgreSQL extensions.

You can install PGMQ directly using PGXN, the PostgreSQL Extension Network, which is a distribution system for open-source PostgreSQL extension libraries.

Install PGMQ using PGXN

```bash

> pgxn install pgmq

Reviewer: Please add hyperlinks to pgxn, either link directly to the pgxn install instructions or put the instructions here.

docker run -d --name postgres -e POSTGRES_PASSWORD=postgres -p 5432:5432 quay.io/tembo/pgmq-pg:latest
```

## PGXN Installation

PGXN is a centralized network to make it easier to install and manage PostgreSQL extensions.

Install PGXN Client:

You can install the PGXN client by following the instructions provided in the [PGXN Client Installation Guide](https://pgxn.github.io/pgxnclient/install.html)

Install PGMQ using PGXN

```bash
pgxn install pgmq
```

Alternatively, you can manually install PGMQ by downloading and building from the source using the following commands:

> 

Author: ```suggestion
> **Note:** Run these commands as the user who owns the PostgreSQL installation (often `root` on Unix-based systems) or use `sudo` if necessary.
```

```
curl -LO https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip
unzip pgmq-1.4.2.zip
cd pgmq-1.4.2
make

> sudo make install

Reviewer 2: I would omit `sudo` here, and suggest instead that you add a comment up top that these commands should be run as the user who owns the Postgres install (e.g., often `root` on *nix systems), or use `sudo`.

Author: Done!

> sudo make install

Author: ```suggestion
make install
```

docker run -d --name postgres -e POSTGRES_PASSWORD=postgres -p 5432:5432 quay.io/tembo/pgmq-pg:latest
```

## PGXN Installation

> PGXN is a centralized network to make it easier to install and manage PostgreSQL extensions.

Reviewer 2: ```suggestion
PGXN is a distributed network of extensions that make it easier to install and manage PostgreSQL extensions.
```

sudo apt-get install postgres-server-dev-14
```

## Platform-Specific Installation Instructions

> ### Windows Installation (Using WSL)

Reviewer: Can we provide instructions for Native windows instead? Instructions to install pgmq on WSL are the same as linux install instructions.

sudo apt-get install postgres-server-dev-14
```

## Platform-Specific Installation Instructions

### Windows Installation

If you're working on native Windows, follow these steps to install and build pgmq:

1. Install PostgreSQL for Windows

- Download the official PostgreSQL installer from [PostgreSQL Windows Downloads](https://www.postgresql.org/download/windows/).

- Follow the installation process, ensuring you include pgAdmin and Command Line Tools

- After installation, add the PostgreSQL binary directory (typically `C:\Program Files\PostgreSQL\<version>\bin`) to your system's `PATH` environment variable. This will allow access to PostgreSQL tools like `pg_config`.

2. Install Build Tools (MinGW)

> - Download and install [MinGW]() from MinGW. Select the GCC compiler for C/C++

Reviewer: @Saturn225 it looks like MinGW is missing the url.

Author: Added!

> - Download and install [MinGW]() from MinGW. Select the GCC compiler for C/C++

Author: ```suggestion
- Download and install [MinGW](https://sourceforge.net/projects/mingw/) from MinGW. Select the GCC compiler for C/C++
```

- During installation, ensure the binaries are added to the `PATH`. MinGW is essential for compiling the `pgmq` extension on Windows.

3.  Install `pgmq` from Source

- Download the `pgmq` source:

```bash
curl -LO https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip
unzip pgmq-1.4.2.zip
cd pgmq-1.4.2
```

- Build and install `pgmq`: Use the following commands to compile and install pgmq using MinGW:

```bash
make PG_CONFIG="C:/Program Files/PostgreSQL/<version>/bin/pg_config"
make install
```

4. Create the pgmq Extension

- After installation, connect to PostgreSQL and create the pgmq extension:

```sql
CREATE EXTENSION pgmq cascade;
```

### Mac Installation

> If you are using macOS, you can install PostgreSQL and build PGMQ using Homebrew.

Reviewer 2: I don't think you need homebrew instructions. One can either use PGXN or download and build manually regardless of how Postgres was installed.

I think I'd suggest providing these categories for installation instruction, after saying the user must already have Postgres and build tools installed:

1. Use the `pgxn` CLI.
2. Download tarball from PGXN and build manually
3. Clone the Git repo and build manually
4. Windows

Author: Yeah! It makes sense

sudo apt-get install postgres-server-dev-14
```

## Platform-Specific Installation Instructions

### Windows Installation

If you're working on native Windows, follow these steps to install and build pgmq:

1. Install PostgreSQL for Windows

- Download the official PostgreSQL installer from [PostgreSQL Windows Downloads](https://www.postgresql.org/download/windows/).

- Follow the installation process, ensuring you include pgAdmin and Command Line Tools

- After installation, add the PostgreSQL binary directory (typically `C:\Program Files\PostgreSQL\<version>\bin`) to your system's `PATH` environment variable. This will allow access to PostgreSQL tools like `pg_config`.

2. Install Build Tools (MinGW)

- Download and install [MinGW](https://sourceforge.net/projects/mingw/) from MinGW. Select the GCC compiler for C/C++
- During installation, ensure the binaries are added to the `PATH`. MinGW is essential for compiling the `pgmq` extension on Windows.

3.  Install `pgmq` from Source

- Download the `pgmq` source:

> ```bash

Author: ```suggestion
```powershell
   Invoke-WebRequest -Uri https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip -OutFile pgmq-1.4.2.zip
   Expand-Archive -Path pgmq-1.4.2.zip -DestinationPath .\pgmq-1.4.2
   cd .\pgmq-1.4.2
```

curl -LO https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip
unzip pgmq-1.4.2.zip

> cd pgmq-1.4.2

Reviewer: @Saturn225 , on windows, should this still be `bash`, and will the user have access to `curl` and `unzip`?

Reviewer: If this is Powershell, lets make that clear.

Reviewer: I'm a noob on Windows, and I'm getting stuck on this part 😹

Author: @ChuckHend Yes that could make different if you go full with Windows Instructions

Author: ```
   Invoke-WebRequest -Uri https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip -OutFile pgmq-1.4.2.zip
   Expand-Archive -Path pgmq-1.4.2.zip -DestinationPath .\pgmq-1.4.2
```

> curl -LO https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip
> unzip pgmq-1.4.2.zip
> cd pgmq-1.4.2

Author: ```suggestion
```

> ```powershell
>    Invoke-WebRequest -Uri https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip -OutFile pgmq-1.4.2.zip
>    Expand-Archive -Path pgmq-1.4.2.zip -DestinationPath .\pgmq-1.4.2
>    cd .\pgmq-1.4.2
> ```

Reviewer 2: ```suggestion
```powershell
Invoke-WebRequest -Uri https://api.pgxn.org/dist/pgmq/1.4.2/pgmq-1.4.2.zip -OutFile pgmq-1.4.2.zip
Expand-Archive -Path pgmq-1.4.2.zip -DestinationPath .\pgmq-1.4.2
cd .\pgmq-1.4.2
```
```
